### DISCUSSION TITLE
Is it possible to set repository properties by using this provider

### DISCUSSION THREAD

**Discussion by jmvitali** [jfrog/terraform-provider-artifactory#806] [General] [answered]
Hello Guys,
Are we able to set local repository properties by using this Artifactory Terraform provider?

**Comment by alexhung** [comment]
@jmvitali Not as far as I can tell. There's no API support to set properties at the repository level. It's only possible to set properties for each artifact in the repository.

**Comment by jmvitali** [comment]
@alexhung,
I confirm we can set the properties of a local repository by using the following REST API

`PUT /api/storage/{repoKey}?properties=p1=v1[,v2][|p2=v3][[&recursive=0]`

**Reply by alexhung** [reply]
Interesting. This looks like an undocumented API? At least I can't find them in the JFrog doc site.

**Reply by jmvitali** [reply]
Do you think the artifactory provider could be improved to add a new ressource related to this rest API?
I'm going to send a request to Jfrog to update documentation to reflect that one.

**Reply by alexhung** [reply, marked as answer]
Yes if:
* API is public to ensure consistency and being maintained
* the API contain CRUD operations. This is required by Terraform.
